### DISCUSSION TITLE
Power-Anzeige in Homeassistant

### DISCUSSION THREAD

**Discussion by mkoxyz** [tbnobody/OpenDTU#1124] [Q&A]
Sorry für die vielleicht doofe Frage, doch irgendwie habe ich keine endgültige Lösung bei meinen Suchen gefunden. Ich habe openDTU erfolgreich in Homeassistant eingebunden. Wo ich allerdings nicht weiterkomme ist die Anzeige der Werte. Ich wollte bei einem Dashboard die aktuelle Leistung der einzelnen Wechselrichter als Messinstrument anzeigen. Das funktioniert super, solange die etwas produzieren. Sobald aber natürlich die Sonne weg ist, bekomme ich natürlich eine Fehlermeldung satt der Anzeige (Messwert ist nicht nummerisch). Bei den Suchen habe ich nur folgende Lösung gefunden, dass man den Wert mit der Funktion "value_template: "{{ states('sensor.your_sensor') | float }}"" umwandeln könnte. So jetzt bin ich nicht wirklich bewandert mit HA und habe versucht, das wie es oben steht (den Namen des sensors geändert) in der configuration.yaml einzubinden, doch das funktioniert nicht, ich bekomme eine Fehlermeldung. Gibt es hierbei eine schönere Lösung? Wie habt Ihr das gelöst?
Danke schon mal für Eure Hilfe.

**Comment by Kampfschildkroete** [comment]
Die Fehlermeldung hilft bei so etwas immer ungemein :)

Ansonsten könnte es helfen, wenn du einen default Wert angibst:
`"value_template: "{{ states('sensorXXX') | float(0) }}""`

Je nachdem was du vorhast, könntest du auch [availabilty](https://www.home-assistant.io/integrations/template/#availability) beim template konfigurieren, z.B.:
`
      availability: >
        {{ states('sensor.XXX') | is_number
`

**Reply by mkoxyz** [reply]
Die Fehlermeldung lautet wie geschrieben, dass der Wert nicht numerisch ist. Ich mache einen Screenshot, sobald einer der Wechselrichter aufhört zu produzieren.
Wie genau muss ich jetzt die Funktion value_template einbinden? Ich nehme an, dass es in der configuration.yaml gemacht wird - allerdings wo genau dort? Damit hatte ich schon Schwierigkeiten bei meiner eigen gesuchten Lösung. ![image](https://github.com/tbnobody/OpenDTU/assets/50030931/5231b8e4-2b4e-4e8c-9ac3-078bad09dcc9)

**Comment by tbnobody** [comment]
> Das funktioniert super, solange die etwas produzieren. Sobald aber natürlich die Sonne weg ist, bekomme ich natürlich eine Fehlermeldung satt der Anzeige (Messwert ist nicht nummerisch).

Wenn dir der Status der Datenpunkte egal ist (also ob der Wert dahinter gerade aktuell ist oder nicht) deaktiviere unter Einstellungen --> MQTT 
![image](https://github.com/tbnobody/OpenDTU/assets/1753222/407de55f-47ab-4272-a139-59fb3cb6ae16)

**Reply by mkoxyz** [reply]
Ich werde dies auch mal testen, allerdings wäre es schon eine schöne Sache, dass der Wert gleich 0 ist, wenn der Wechselrichter nichts produziert.

**Reply by mkoxyz** [reply]
Das hat soweit sehr gut funktioniert. Die Werte lagen sogar bei 0. Super, danke!
